rustainers/src/runner/inner.rs

```diff
@@ -40,6 +41,14 @@ pub(crate) trait InnerRunner: Display + Debug + Send + Sync {
         Ok(result)
     }
 
+    #[tracing::instrument(level = "debug", skip(self), fields(runner = %self))]
+    async fn list_networks(&self) -> Result<Vec<RunnerNetwork>, ContainerError> {
+        let mut cmd = self.command();
+        cmd.push_args(["network", "ls", "--no-trunc", "--format={{json .}}"]);
+        let result = cmd.json_stream::<RunnerNetwork>().await?;
+        Ok(result)
+    }
```

Comment:
I guess that you don't want to list all networks but the one used by the container. So may you should use `inspect` (and enhance existing related function)

Reply:
Actually, I want to list all the network for a given runner and not just for a given container.
The intent is to have the ability to run a container inside another container created in the context of a given network. That allows us to solve the docker in docker for github ci